keys/signer_ed25519.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2020 Coinbase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package keys
+
+import (
+	"crypto/ed25519"
+	"fmt"
+
+	"github.com/coinbase/rosetta-sdk-go/asserter"
+	"github.com/coinbase/rosetta-sdk-go/types"
+)
+
+// SignerEd25519 is initialized from a keypair
+type SignerEd25519 struct {
+	KeyPair *KeyPair
+}
+
+// PublicKey returns the PublicKey of the signer
+func (s SignerEd25519) PublicKey() *types.PublicKey {
+	return s.KeyPair.PublicKey
+}
+
+// Signs arbitrary payloads using a KeyPair
+func (s SignerEd25519) Sign(payload *types.SigningPayload, sigType types.SignatureType) (*types.Signature, error) {
+	err := s.KeyPair.IsValid()
+	if err != nil {
+		return nil, err
+	}
+
+	if !(payload.SignatureType == types.Ed25519 || payload.SignatureType == "") {
+		return nil, fmt.Errorf("sign: payload signature type is not ed25519")
```

Comment:
nit: you may want to use the constant in the error message instead of typing out `ed25519`...same can be done on 47